assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java

```diff
@@ -3514,6 +3499,76 @@ public AssignmentSubmission getSubmission(String assignmentReference, User perso
 		return submission;
 	}
 
+	/** 
+	 * Gets a map of users to their submissions for the specified assignment
+	 * @param a the assignment in question
+	 * @param users the users making up the key set
+	 */
+	public Map<User, AssignmentSubmission> getUserSubmissionMap(Assignment a, List<User> users)
+	{
+		Map<User, AssignmentSubmission> userSubmissionMap = new HashMap<>();
+		if (a != null && !CollectionUtils.isEmpty(users))
+		{
+			if (a.isGroup())
+			{
+				userSubmissionMap.putAll(getUserGroupSubmissionMap(a, users));
+			}
+			else
+			{
+				// SAK-31130 - get all submissions for these users with a single query
```

Comment:
You don't need to reference the SAK ticket number in comments. Git blame can tell us this info. Same deal in a few other places on this PR.